File: data_structures/linkedlists_py/lec2/revK.py
class Node: 
    def __init__(self, data, next=None):
        self.data = data 
        self.next = next 

def printLL (head: Node):
    while (head != None):
        print(head.data, "->", end="")
        head = head.next 

def revLL(head, k):
    temp = head 
    count = 0
    while (temp != None or count == k):
        temp = temp.next 
        count += 1
        
        

File: data_structures/linkedLists/lec3/revK.cpp
#include <iostream>
#include <cmath>

using namespace std;

class Node
{
public:
    int data;
    Node *next;

public:
    Node(int x, Node *next1)
    {
        data = x;
        next = next1;
    }

    Node(int x)
    {
        data = x;
        next = nullptr;
    }
};

Node *kReverse(Node *head, int k);

Node *getKthNode(Node *head, int k)
{
    Node *temp = head;
    while (count != k && head != nullptr)
    {
        count++;
        head = head->next;
    }
    return head;
}

Node *reverseLL(Node *head)
{
    Node *prev = nullptr;
    Node *curr = head;

    while (curr != nullptr)
    {
        Node *temp = curr->next;
        curr->next = prev;
        prev = curr;
        curr = temp;
    }

    return prev;
}

int main()
{
    return 0;
}